compiler/rustc_hir_analysis/src/coherence/orphan.rs

```diff
@@ -452,7 +452,6 @@ fn lint_auto_trait_impl<'tcx>(
     trait_ref: ty::TraitRef<'tcx>,
     impl_def_id: LocalDefId,
 ) {
-    assert_eq!(trait_ref.args.len(), 1);
```

Comment:
Please turn this `assert_eq` into a `delay_span_bug`, and remove the `&& trait_ref.args.len() == 1` above. I don't think we need both.